coderd/templates.go

```diff
@@ -552,6 +552,83 @@ func (api *API) templateDAUs(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, resp)
 }
 
+// @Summary Get template app usage by ID
+// @ID get-template-app-usage-by-id
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Templates
+// @Param template path string true "Template ID" format(uuid)
+// @Param from query string false "Filter by app usage starting from this date"
+// @Param to query string false "Filter by app usage until this date"
+// @Success 200 {object} codersdk.TemplateAppUsageResponse
+// @Router /templates/{template}/app-usage [get]
+func (api *API) appUsage(rw http.ResponseWriter, r *http.Request) {
+	var err error
+	ctx := r.Context()
+	template := httpmw.TemplateParam(r)
+	if !api.Authorize(r, rbac.ActionRead, template) {
+		httpapi.ResourceNotFound(rw)
+		return
+	}
+
+	// Default date filter values
+	since := time.Now().Add(-24 * 7 * time.Hour) // Default is last 7 days
+	to := time.Now()
+
+	// Assign date filter values
+	queryParamsValidation := []codersdk.ValidationError{}
+	q := r.URL.Query()
+	sinceQueryStr := q.Get("since")
```

Comment:
IMO the query parameters should be "from" and "to", or "since" and "until" so they match

Reply:
Sounds good